### PR TITLE
unifying: notify users to replug when using dfu control (Fixes: #1383)

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
I pushed back to the original message because I'm expecting by the false return we do get a disjointed process now that user gets a message, acts and tries it again.

Looking forward to testing results to confirm that though.

**Comment by superm1** [comment]
btw if this is the approach we end up going, I think we should either:
1. make these strings translatable things that the plugins produce (frontends would just show the translated string when they rendered an error)
or
2. make a new error type (frontends will need to recognize the new error type and carry translations for it)

**Comment by FFY00** [comment]
Hey, sorry for the late response. I am back in Portugal in college and this week has been just a tiny bit chaotic. I hope to review this by the weekend but I can't promise anything as I am supposed to be studying full time. Thanks for the understanding :)

**Comment by superm1** [comment]
Thanks for checking in.  I don't have any of these peripherals that support it so I can't check it myself.  @hughsie do you have anything you can check it with?

**Comment by hughsie** [comment]
> hughsie do you have anything you can check it with

Nope, I've only got the auto-restart stuff.

**Comment by superm1** [comment]
I made some changes to how 8bitdo works with a new error type.  Now this should cause the error message to get cached in `UpdateError` and a client can potentially show this message to a user between runs.

**Comment by superm1** [comment]
Now in between steps this is how the device populates:
```
├─8Bitdo SF30 Pro:
│     Device ID:           672c087de09848d9e7ee32aa1dea2fbeb8b81e6b
│     Summary:             A redesigned classic game controller
│     Current version:     1.23
│     Vendor:              8Bitdo (USB:0x2DC8)
│     Update State:        failed-transient
│     Update Error:        Device 8Bitdo SF30 Pro needs to manually be put in update mode: Unplug the controller, hold down L1 + R1 + START until the red LED on top of the controller is flashing and then reconnect the controller.
│     GUIDs:               043c7673-bd45-5e2e-b8b9-c22df2dfec2a
│                          b02f1319-c1c4-54f4-b7f2-5bb3552f6a9e
│                          e95bfb47-11f9-5afd-856e-fbc45730456d
│                          269b3121-097b-50d8-b9ba-d1f64f9cd241
│     Device Flags:        • Updatable
│                          • Supported on remote server
│                          • Requires a bootloader
│                          • Device will not re-appear after update completes
```

**Comment by superm1** [comment]
@hughsie can you take a look at this again?  If the general and 8bitdo related stuff make sense to you, I think we should adopt the same thing for these other peripherals (even if neither of us can test them).  They will then at least behave the same as 8bitdo does.

**Comment by FFY00** [comment]
For some reason I can't get the device to actually show up.

**Comment by superm1** [comment]
> For some reason I can't get the device to actually show up. @hughsie any ideas?

Have you updated to current master? If you're missing https://github.com/fwupd/fwupd/commit/45a181558777836ff51d2360a37d6b710c0bca50 but have https://github.com/fwupd/fwupd/commit/51a869a01da445cd77dd9baddcf05d3bf0b4debf might explain.

Otherwise, `fwupdtool --plugin-whitelist=unifying --verbose` output would be good.

**Comment by FFY00** [comment]
I rebased on master and it's still the same.

```
$ sudo src/fwupdtool get-updates --plugin-whitelist=unifying --verbose
00:22:11:0349 FuDebug              Verbose debugging enabled (on console 1)
00:22:11:0351 FuMain               Failed to stop daemon: failed to find fwupd.service: GDBus.Error:org.freedesktop.systemd1.NoSuchUnit: Unit fwupd.service not loaded.
00:22:11:0351 FuConfig             loading config values from /etc/fwupd/daemon.conf
00:22:11:0352 GLib-GIO             _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
00:22:11:0352 Fwupd                looking for os-release at /etc/os-release
00:22:11:0352 FuConfig             loading /usr/share/fwupd/metainfo
00:22:11:0352 XbSilo               attempting to load /var/cache/fwupd/metainfo.xmlb
00:22:11:0353 XbSilo               file: 107514d9-8bf4-5534-9f80-0c0808a08b56, current:107514d9-8bf4-5534-9f80-0c0808a08b56, cached: (null)
00:22:11:0353 XbSilo               loading silo with file contents
00:22:11:0353 FuConfig             using config path of /etc/fwupd
00:22:11:0353 FuConfig             skipping invalid file /etc/fwupd/remotes.d/lvfs.conf.pacsave
00:22:11:0353 FuConfig             skipping invalid file /etc/fwupd/remotes.d/lvfs-testing.conf.pacsave
00:22:11:0353 FuConfig             skipping invalid file /etc/fwupd/remotes.d/lvfs-testing.conf.pacnew
00:22:11:0353 FuConfig             skipping invalid file /etc/fwupd/remotes.d/lvfs.conf.pacnew
00:22:11:0353 FuConfig             loading config from /etc/fwupd/remotes.d/lvfs.conf
00:22:11:0353 FuConfig             loading config from /etc/fwupd/remotes.d/lvfs-testing.conf
00:22:11:0353 FuConfig             loading config from /etc/fwupd/remotes.d/vendor.conf
00:22:11:0353 FuConfig             loading config from /etc/fwupd/remotes.d/vendor-directory.conf
00:22:11:0353 FuConfig             loading config from /etc/fwupd/remotes.d/fwupd-tests.conf
00:22:11:0353 FuConfig             loading config from /etc/fwupd/remotes.d/dell-esrt.conf
00:22:11:0353 FuConfig             using config path of /usr/lib/fwupd
00:22:11:0353 FuConfig             path /usr/lib/fwupd/remotes.d does not exist
00:22:11:0353 FuConfig             ignoring unfound remote fwupd
00:22:11:0353 FuConfig             ordering lvfs-testing=lvfs+1
00:22:11:0353 FuConfig             ignoring unfound remote fwupd
00:22:11:0353 FuConfig             ignoring unfound remote fwupd
00:22:11:0353 FuConfig             ignoring unfound remote fwupd
00:22:11:0358 FuEngine             client certificate exists and working
00:22:11:0358 FuHistory            trying to open database '/var/lib/fwupd/pending.db'
00:22:11:0358 FuHistory            got schema version of 5
00:22:11:0358 FuHwids              smbios property Manufacturer=ASUS
00:22:11:0358 FuHwids              smbios property EnclosureKind=3
00:22:11:0358 FuHwids              smbios property Family=ASUS MB
00:22:11:0358 FuHwids              smbios property ProductName=All Series
00:22:11:0358 FuHwids              smbios property ProductSku=All
00:22:11:0358 FuHwids              smbios property BiosVendor=American Megatrends Inc.
00:22:11:0358 FuHwids              smbios property BiosVersion=2906
00:22:11:0358 FuHwids              smbios property BiosMajorRelease=04
00:22:11:0358 FuHwids              smbios property BiosMinorRelease=06
00:22:11:0358 FuHwids              smbios property BaseboardManufacturer=ASUSTeK COMPUTER INC.
00:22:11:0358 FuHwids              smbios property BaseboardProduct=H97-PRO
00:22:11:0361 FuQuirks             no /var/lib/fwupd/quirks.d, skipping
00:22:11:0361 XbSilo               attempting to load /var/cache/fwupd/quirks.xmlb
00:22:11:0361 XbSilo               file: 5b1bf00c-9438-5400-8c2d-cd5c46df33f5, current:5b1bf00c-9438-5400-8c2d-cd5c46df33f5, cached: (null)
00:22:11:0361 XbSilo               loading silo with file contents
00:22:11:0361 FuEngine             remote lvfs-testing not enabled, so skipping
00:22:11:0361 FuEngine             remote fwupd-tests not enabled, so skipping
00:22:11:0361 FuEngine             remote vendor not enabled, so skipping
00:22:11:0361 FuEngine             remote vendor-directory not enabled, so skipping
00:22:11:0361 XbSilo               attempting to load /var/cache/fwupd/metadata.xmlb
00:22:11:0361 XbSilo               0.01ms      parse blob
00:22:11:0361 XbSilo               0.03ms      loaded file
00:22:11:0361 XbSilo               file: 7f36bb69-064a-5fba-840a-9bf9160b0e1e, current:7f36bb69-064a-5fba-840a-9bf9160b0e1e, cached: (null)
00:22:11:0361 XbSilo               loading silo with file contents
00:22:11:0361 XbSilo               0.00ms      parse blob
00:22:11:0361 XbSilo               0.12ms      query on / with `components/component` limit=0 -> 307 results
00:22:11:0361 FuEngine             307 components now in silo
00:22:11:0361 XbSilo               0.16ms      query on / with `components/component/provides/firmware` limit=0 -> 366 results
00:22:11:0361 XbSilo               0.07ms      query on / with `components/component/provides/firmware` limit=0 -> 366 results
00:22:11:0366 FuEngine             plugin udev is not whitelisted
00:22:11:0366 FuEngine             plugin ata is not whitelisted
00:22:11:0366 FuEngine             plugin dfu is not whitelisted
00:22:11:0366 FuEngine             plugin colorhug is not whitelisted
00:22:11:0366 FuEngine             plugin coreboot is not whitelisted
00:22:11:0366 FuEngine             plugin ebitdo is not whitelisted
00:22:11:0366 FuEngine             plugin fastboot is not whitelisted
00:22:11:0366 FuEngine             plugin jabra is not whitelisted
00:22:11:0366 FuEngine             plugin steelseries is not whitelisted
00:22:11:0366 FuEngine             plugin dell_dock is not whitelisted
00:22:11:0366 FuEngine             plugin nitrokey is not whitelisted
00:22:11:0366 FuEngine             plugin optionrom is not whitelisted
00:22:11:0366 FuEngine             plugin rts54hid is not whitelisted
00:22:11:0366 FuEngine             plugin rts54hub is not whitelisted
00:22:11:0366 FuEngine             plugin solokey is not whitelisted
00:22:11:0366 FuEngine             plugin synaptics_cxaudio is not whitelisted
00:22:11:0366 FuEngine             plugin synaptics_prometheus is not whitelisted
00:22:11:0366 FuEngine             plugin thelio_io is not whitelisted
00:22:11:0366 FuEngine             adding plugin /usr/lib/fwupd-plugins-3/libfu_plugin_unifying.so
00:22:11:0366 FuPlugin             performing init() on /usr/lib/fwupd-plugins-3/libfu_plugin_unifying.so
00:22:11:0366 FuPlugin             added udev subsystem watch of hidraw
00:22:11:0366 FuEngine             plugin upower is not whitelisted
00:22:11:0366 FuEngine             plugin wacom_raw is not whitelisted
00:22:11:0366 FuEngine             plugin wacom_usb is not whitelisted
00:22:11:0366 FuEngine             plugin superio is not whitelisted
00:22:11:0366 FuEngine             plugin synaptics_rmi is not whitelisted
00:22:11:0366 FuEngine             plugin vli_usbhub is not whitelisted
00:22:11:0366 FuEngine             plugin csr is not whitelisted
00:22:11:0366 FuEngine             plugin emmc is not whitelisted
00:22:11:0366 FuEngine             plugin nvme is not whitelisted
00:22:11:0366 FuEngine             plugin altos is not whitelisted
00:22:11:0366 FuEngine             plugin amt is not whitelisted
00:22:11:0366 FuEngine             plugin thunderbolt is not whitelisted
00:22:11:0366 FuEngine             plugin thunderbolt_power is not whitelisted
00:22:11:0366 FuEngine             plugin redfish is not whitelisted
00:22:11:0366 FuEngine             plugin dell is not whitelisted
00:22:11:0366 FuEngine             plugin dell_esrt is not whitelisted
00:22:11:0366 FuEngine             plugin synapticsmst is not whitelisted
00:22:11:0366 FuEngine             plugin uefi is not whitelisted
00:22:11:0366 FuEngine             plugin uefi_recovery is not whitelisted
00:22:11:0366 FuEngine             Emitting PropertyChanged('Status'='loading')
Loading…                 [-                                      ]00:22:11:0366 FuPlugin             performing startup() on unifying
00:22:11:0366 FuEngine             using plugins: unifying
00:22:11:0381 FuEngine             14 devices with subsystem hidraw
00:22:11:0382 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0382 XbSilo               0.19ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='40f213ea-bbbf-5f0b-b11d-fb3536a05f3d']` limit=1 -> 0 results
00:22:11:0382 XbSilo               0.16ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='72138597-f997-59b1-ab95-b7afc329f388']` limit=1 -> 0 results
00:22:11:0382 FuPlugin             40f213ea-bbbf-5f0b-b11d-fb3536a05f3d,72138597-f997-59b1-ab95-b7afc329f388 has no updates, so ignoring device
00:22:11:0383 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0383 XbSilo               0.16ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='40f213ea-bbbf-5f0b-b11d-fb3536a05f3d']` limit=1 -> 0 results
00:22:11:0383 XbSilo               0.15ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='72138597-f997-59b1-ab95-b7afc329f388']` limit=1 -> 0 results
00:22:11:0383 FuPlugin             40f213ea-bbbf-5f0b-b11d-fb3536a05f3d,72138597-f997-59b1-ab95-b7afc329f388 has no updates, so ignoring device
00:22:11:0384 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0385 XbSilo               0.15ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='b3e64459-43c7-5ec8-9771-e1223093b2ce']` limit=1 -> 0 results
00:22:11:0385 XbSilo               0.15ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='dc6907b5-2d37-559b-b0c5-b46f29214668']` limit=1 -> 0 results
00:22:11:0385 FuPlugin             b3e64459-43c7-5ec8-9771-e1223093b2ce,dc6907b5-2d37-559b-b0c5-b46f29214668 has no updates, so ignoring device
00:22:11:0385 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0386 XbSilo               0.18ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='b3e64459-43c7-5ec8-9771-e1223093b2ce']` limit=1 -> 0 results
00:22:11:0386 XbSilo               0.16ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='dc6907b5-2d37-559b-b0c5-b46f29214668']` limit=1 -> 0 results
00:22:11:0386 FuPlugin             b3e64459-43c7-5ec8-9771-e1223093b2ce,dc6907b5-2d37-559b-b0c5-b46f29214668 has no updates, so ignoring device
00:22:11:0386 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0387 XbSilo               0.20ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='7207c83d-42ff-5742-a410-6244feb8fc8c']` limit=1 -> 0 results
00:22:11:0387 XbSilo               0.48ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='a084c0da-57f9-54c1-8f1b-f2515c0c04ce']` limit=1 -> 0 results
00:22:11:0387 FuPlugin             7207c83d-42ff-5742-a410-6244feb8fc8c,a084c0da-57f9-54c1-8f1b-f2515c0c04ce has no updates, so ignoring device
00:22:11:0388 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0388 XbSilo               0.18ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='b3e64459-43c7-5ec8-9771-e1223093b2ce']` limit=1 -> 0 results
00:22:11:0389 XbSilo               0.16ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='dc6907b5-2d37-559b-b0c5-b46f29214668']` limit=1 -> 0 results
00:22:11:0389 FuPlugin             b3e64459-43c7-5ec8-9771-e1223093b2ce,dc6907b5-2d37-559b-b0c5-b46f29214668 has no updates, so ignoring device
00:22:11:0389 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0389 XbSilo               0.15ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='bcab588c-1014-5fc6-b123-2e125aa3d9da']` limit=1 -> 0 results
00:22:11:0390 XbSilo               0.15ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='9bb93661-fefc-5522-8216-50f6d3e29def']` limit=1 -> 0 results
00:22:11:0390 FuPlugin             bcab588c-1014-5fc6-b123-2e125aa3d9da,9bb93661-fefc-5522-8216-50f6d3e29def has no updates, so ignoring device
00:22:11:0390 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0391 XbSilo               0.14ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='168fa758-b89a-5e33-b12f-533b7842beea']` limit=1 -> 0 results
00:22:11:0391 XbSilo               0.15ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='d8f438d3-5bdb-5175-8be7-97f58c7fa59e']` limit=1 -> 0 results
00:22:11:0391 FuPlugin             168fa758-b89a-5e33-b12f-533b7842beea,d8f438d3-5bdb-5175-8be7-97f58c7fa59e has no updates, so ignoring device
00:22:11:0391 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0392 XbSilo               0.15ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='354d8012-b882-544e-bc91-54fddf7b0d6e']` limit=1 -> 0 results
00:22:11:0392 XbSilo               0.14ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='6e7b2a0c-d7a1-5d16-928e-733e42207fde']` limit=1 -> 0 results
00:22:11:0392 FuPlugin             354d8012-b882-544e-bc91-54fddf7b0d6e,6e7b2a0c-d7a1-5d16-928e-733e42207fde has no updates, so ignoring device
00:22:11:0392 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0393 XbSilo               0.14ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='e0979ae2-dcd4-55ea-81b1-7508cc6b0ea1']` limit=1 -> 0 results
00:22:11:0393 XbSilo               0.14ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='f132524b-dca0-5ea4-9f59-fde02c7361ac']` limit=1 -> 0 results
00:22:11:0393 FuPlugin             e0979ae2-dcd4-55ea-81b1-7508cc6b0ea1,f132524b-dca0-5ea4-9f59-fde02c7361ac has no updates, so ignoring device
00:22:11:0394 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0394 XbSilo               0.15ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='3213fa8f-d742-5f60-a65a-26b0855e4bde']` limit=1 -> 0 results
00:22:11:0394 XbSilo               0.14ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='2970c277-33de-5c09-bb6c-9c54a8fc08d1']` limit=1 -> 0 results
00:22:11:0394 FuPlugin             3213fa8f-d742-5f60-a65a-26b0855e4bde,2970c277-33de-5c09-bb6c-9c54a8fc08d1 has no updates, so ignoring device
00:22:11:0395 FuDevice             using 279ed287-3607-549e-bacc-f873bb9838c4 for HIDRAW\VEN_046D&DEV_C52B
00:22:11:0395 XbSilo               0.15ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='32a13a86-b689-5f83-93b3-9faebdd938e8']` limit=1 -> 0 results
00:22:11:0395 XbSilo               0.14ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='0c105c2f-03f8-5321-996d-517e43d85f6e']` limit=1 -> 0 results
00:22:11:0395 FuPlugin             32a13a86-b689-5f83-93b3-9faebdd938e8,0c105c2f-03f8-5321-996d-517e43d85f6e has no updates, so ignoring device
00:22:11:0396 FuDevice             already set GType to FuUnifyingRuntime, ignoring FuUnifyingPeripheral
00:22:11:0396 FuDevice             already set GType to FuUnifyingRuntime, ignoring FuUnifyingRuntime
00:22:11:0396 FuDevice             already set GType to FuUnifyingRuntime, ignoring FuUnifyingPeripheral
00:22:11:0396 FuDevice             using f7c450aa2530c34b779d10cd15614e4318ece837 for DEVPATH=/devices/pci0000:00/0000:00:14.0/usb2/2-9/2-9.3/2-9.3.4/2-9.3.4:1.2
00:22:11:0604 FuPlugin             emit added from unifying: f7c450aa2530c34b779d10cd15614e4318ece837
00:22:11:0604 FuDeviceList         ::added f7c450aa2530c34b779d10cd15614e4318ece837
00:22:11:0604 FuMain               ADDED:
FuUnifyingRuntime:
Unifying Receiver
  DeviceId:             f7c450aa2530c34b779d10cd15614e4318ece837
  Guid:                 cc4cbfa9-bf9d-540b-b92b-172ce31013c1
  Guid:                 279ed287-3607-549e-bacc-f873bb9838c4 <- HIDRAW\VEN_046D&DEV_C52B
  Summary:              A miniaturised USB wireless receiver
  Plugin:               unifying
  Flags:                updatable|registered
  Vendor:               Logitech, Inc.
  VendorId:             USB:0x046D
  Version:              RQR24.11_B0036
  VersionBootloader:    BOT03.02_B0008
  VersionFormat:        plain
  Icon:                 preferences-desktop-keyboard
  InstallDuration:      30
  Created:              2019-11-01
  PhysicalId:            DEVPATH=/devices/pci0000:00/0000:00:14.0/usb2/2-9/2-9.3/2-9.3.4/2-9.3.4:1.2
  SignedFirmware:        true

00:22:11:0605 XbSilo               0.59ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='cc4cbfa9-bf9d-540b-b92b-172ce31013c1']/../..|components/component/provides/firmware[@type='flashed'][text()='279ed287-3607-549e-bacc-f873bb9838c4']/../..` limit=1 -> 1 results
00:22:11:0605 FuHistory            get device
00:22:11:0606 FuDevice             already set GType to FuUnifyingRuntime, ignoring FuUnifyingRuntime
00:22:11:0606 FuDevice             already set GType to FuUnifyingRuntime, ignoring FuUnifyingPeripheral
00:22:11:0606 FuDevice             using f7c450aa2530c34b779d10cd15614e4318ece837 for DEVPATH=/devices/pci0000:00/0000:00:14.0/usb2/2-9/2-9.3/2-9.3.4/2-9.3.4:1.2
00:22:11:0814 FuPlugin             emit added from unifying: f7c450aa2530c34b779d10cd15614e4318ece837
00:22:11:0814 FuDeviceList         device f7c450aa2530c34b779d10cd15614e4318ece837 already exists, ignoring
00:22:11:0815 XbSilo               0.40ms      query on / with `components/component/provides/firmware[@type='flashed'][text()='cc4cbfa9-bf9d-540b-b92b-172ce31013c1']/../..|components/component/provides/firmware[@type='flashed'][text()='279ed287-3607-549e-bacc-f873bb9838c4']/../..` limit=1 -> 1 results
00:22:11:0815 FuHistory            get device
00:22:11:0815 FuCommon             not checking RQR24.07_B0030 as no version format set
00:22:11:0815 FuEngine             Emitting PropertyChanged('Status'='idle')
Loading…                 [***************************************]
00:22:11:0816 XbSilo               ignoring for OR statement: indexed string '279ed287-3607-549e-bacc-f873bb9838c4' was unfound
00:22:11:0816 XbSilo               0.39ms      query on / with `components/component/provides/firmware[@type=$'flashed'][text()=$'cc4cbfa9-bf9d-540b-b92b-172ce31013c1']/../..|components/component/provides/firmware[@type=$'flashed'][text()=$'279ed287-3607-549e-bacc-f873bb9838c4']/../..` limit=0 -> 2 results
00:22:11:0816 XbSilo               0.02ms      query on component with `provides/firmware[@type='flashed']` limit=0 -> 2 results
00:22:11:0816 XbSilo               0.00ms      query on component with `releases/release` limit=1 -> 1 results
00:22:11:0816 XbSilo               0.02ms      query on component with `custom/value[@key='LVFS::VersionFormat']` limit=1 -> 0 results
00:22:11:0816 XbSilo               0.01ms      query on release with `checksum[@target='content']` limit=1 -> 1 results
00:22:11:0816 FuKeyring            firmware archive contained no signature
00:22:11:0816 XbSilo               0.01ms      query on component with `requires/*` limit=0 -> 4 results
00:22:11:0816 FuEngine             requirement 0.8.2 ge 1.3.3 on org.freedesktop.fwupd passed
00:22:11:0816 XbSilo               0.02ms      query on component with `provides/firmware[@type='flashed']` limit=0 -> 4 results
00:22:11:0816 XbSilo               0.00ms      query on component with `releases/release` limit=1 -> 1 results
00:22:11:0816 XbSilo               0.02ms      query on component with `custom/value[@key='LVFS::VersionFormat']` limit=1 -> 0 results
00:22:11:0816 XbSilo               0.02ms      query on release with `checksum[@target='content']` limit=1 -> 1 results
00:22:11:0816 FuKeyring            firmware archive contained no signature
00:22:11:0816 XbSilo               0.01ms      query on component with `requires/*` limit=0 -> 4 results
00:22:11:0816 FuEngine             requirement 1.3.1 ge 1.3.3 on org.freedesktop.fwupd passed
00:22:11:0816 XbSilo               0.00ms      query on component with `releases/release` limit=0 -> 1 results
00:22:11:0816 XbSilo               0.00ms      query on component with `id` limit=1 -> 1 results
00:22:11:0816 XbSilo               0.02ms      query on component with `url[@type='homepage']` limit=1 -> 1 results
00:22:11:0816 XbSilo               0.00ms      query on component with `project_license` limit=1 -> 1 results
00:22:11:0816 XbSilo               0.00ms      query on component with `name` limit=1 -> 1 results
00:22:11:0816 XbSilo               0.00ms      query on component with `summary` limit=1 -> 1 results
00:22:11:0816 XbSilo               0.00ms      query on component with `developer_name` limit=1 -> 1 results
00:22:11:0816 XbSilo               0.03ms      query on component with `../custom/value[@key='fwupd::RemoteId']` limit=1 -> 1 results
00:22:11:0816 XbSilo               0.00ms      query on release with `description` limit=1 -> 1 results
00:22:11:0817 XbSilo               0.00ms      query on release with `location` limit=1 -> 1 results
00:22:11:0817 XbSilo               0.01ms      query on release with `checksum[@target='content']` limit=1 -> 1 results
00:22:11:0817 XbSilo               0.01ms      query on release with `url[@type='details']` limit=1 -> 0 results
00:22:11:0817 XbSilo               0.02ms      query on release with `url[@type='source']` limit=1 -> 0 results
00:22:11:0817 XbSilo               0.02ms      query on release with `checksum[@target='container']` limit=1 -> 1 results
00:22:11:0817 XbSilo               0.01ms      query on release with `size[@type='installed']` limit=1 -> 1 results
00:22:11:0817 XbSilo               0.00ms      query on component with `screenshots/screenshot/caption` limit=1 -> 0 results
00:22:11:0817 XbSilo               0.00ms      query on component with `screenshots/screenshot/image` limit=1 -> 0 results
00:22:11:0817 XbSilo               0.01ms      query on component with `custom/value[@key='LVFS::UpdateProtocol']` limit=1 -> 0 results
00:22:11:0817 XbSilo               0.02ms      query on component with `custom/value[@key='LVFS::UpdateMessage']` limit=1 -> 0 results
00:22:11:0817 FuEngine             ignoring RQR24.11_B0036 as the same as RQR24.11_B0036
No upgrades for Unifying Receiver, current is RQR24.11_B0036: RQR24.11_B0036=same
```

**Comment by superm1** [comment]
Try setting `FWUPD_PROBE_VERBOSE=1` in the environment and run again.  I guess the device is in `lsusb` (or otherwise working?)

**Comment by FFY00** [comment]
Ah, no. It's a wireless device connected to the receiver. fwupd should enumerate the connected devices when it sees the receiver. Eh, actually `hid-logitech-dj` does export a hidraw node for the device. IIRC fwupd enumerates the devices using sysfs? I am not really sure. Richard should be able to clear things up.

File: src/fu-engine.c [hunk 1]
```diff
@@ -1611,16 +1611,18 @@ fu_engine_install (FuEngine *self,
...
 		} else {
 			g_set_error (error,
 				     FWUPD_ERROR,
-				     FWUPD_ERROR_INTERNAL,
+				     FWUPD_ERROR_NEEDS_USER,
 				     "Device %s needs to manually be put in update mode",
 				     fu_device_get_name (device));
 		}
+		fu_device_set_update_state (device, FWUPD_UPDATE_STATE_FAILED_TRANSIENT);
+		fu_device_set_update_error (device, (*error)->message);
```
Comment (hughsie) [inline src/fu-engine.c]: This isn't good practice, as it's perfectly valid to set error to NULL in the caller. You need to do the error_local dance here.

File: libfwupd/fwupd-error.h [hunk 2]
```diff
@@ -30,6 +30,7 @@ G_BEGIN_DECLS
  * @FWUPD_ERROR_PERMISSION_DENIED:		Permission was denied
  * @FWUPD_ERROR_BROKEN_SYSTEM:			User has configured their system in a broken way
  * @FWUPD_ERROR_BATTERY_LEVEL_TOO_LOW:		The system battery level is too low
+ * @FWUPD_ERROR_NEEDS_USER:			User needs to do an action to complete the update
```
Comment (hughsie) [inline libfwupd/fwupd-error.h]: `NEEDS_USER` doesn't mean much. `NEEDS_USER_ACTION` might be better, surely?

Comment (superm1) [reply libfwupd/fwupd-error.h]: Will adjust

File: src/fu-engine.c (hunk 1 above)
Comment (superm1) [reply src/fu-engine.c]: It's not though for this function.  Look 20 lines above and you'll see:
```	g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
```

**Comment by superm1** [comment]
> Ah, no. It's a wireless device connected to the receiver. fwupd should enumerate the connected devices when it sees the receiver. Eh, actually `hid-logitech-dj` does export a hidraw node for the device. IIRC fwupd enumerates the devices using sysfs? I am not really sure. Richard should be able to clear things up.

Is this more unintentional fallout on hidraw nodes from the move to Xmlb for quirks perhaps?

**Comment by FFY00** [comment]
> Is this more unintentional fallout on hidraw nodes from the move to Xmlb for quirks perhaps?

I am not sure, we should have a test for this. With a little bit of work I think I could get https://github.com/libratbag/ratbag-emu to emulate this devices.

File: src/fu-engine.c (hunk 1 above)
Comment (hughsie) [reply src/fu-engine.c]: > g_return_val_if_fail (error == NULL || *error == NULL, FALSE);

If `error` is NULL then we don't return with failure. This is checking that either error is NULL, *or* *error is NULL and this unset.

Comment (superm1) [reply src/fu-engine.c]: OK just pushed a fixup to test.  I'm going to be AFK for a while in meetings.  Feel free to fixup yourself if any other changes needed to get this into 1.3.3.

**Pull request merged**